### DISCUSSION TITLE
`npm` in Github Actions not able to access private repository dependency

### DISCUSSION THREAD

**Discussion by ArshanKhanifar** [community/community#26638] [Actions]
<p>I have a react native app that uses a native module which is in a different repository. Both my application and the module library are in private repos on Github, under my company’s organization. But both are private repos.</p>
<p>I have the module listed in my <code>package.json</code> file like so:</p>
<pre><code class="lang-auto">  "dependencies": {
    "react-native-cellular-info": "https://github.com/repeaterstore/react-native-cellular-info",
  }
</code></pre>
<p>And I have a <strong>Github Action</strong> that checks out the repository and does a <code>yarn install</code>:</p>
<pre><code class="lang-auto">      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
      - run: yarn install
</code></pre>
<p>But I get this message:</p>
<pre><code class="lang-auto">npm verb npm  v6.14.12
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t https://github.com/repeaterstore/react-native-cellular-info.git
npm ERR! 
npm ERR! remote: Repository not found.
npm ERR! fatal: repository 'https://github.com/repeaterstore/react-native-cellular-info.git/' not found
npm ERR! 
npm ERR! exited with error code: 128
npm verb exit [ 1, true ]
npm timing npm Completed in 569ms
</code></pre>
<p>At first I thought this was an authorization issue, and used <a href="https://github.com/fusion-engineering/setup-git-credentials" rel="noopener nofollow ugc">this action</a> along with an access token to ensure that the container has access to my private repos. My configuration now looks like:</p>
<pre><code class="lang-auto">    steps:
      - name: Setup git credentials
        uses: fusion-engineering/setup-git-credentials@v2
        with:
          credentials: ${{secrets.GIT_CREDENTIALS}}
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
      # other stuff...
</code></pre>
<p>I’m even able to clone the private repository now:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/9/899210e765d5af3a1ec2bc1a408051ba59cccc72.png" data-download-href="/uploads/short-url/jD0mdZF1qKIGVkPqpy7UM7pA6MG.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/10492324/181099003-8dde7920-a164-454a-95c8-395ea5d65443.png" alt="image" data-base62-sha1="jD0mdZF1qKIGVkPqpy7UM7pA6MG" width="690" height="328" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/9/899210e765d5af3a1ec2bc1a408051ba59cccc72_2_690x328.png, https://aws1.discourse-cdn.com/github/original/3X/8/9/899210e765d5af3a1ec2bc1a408051ba59cccc72.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/8/9/899210e765d5af3a1ec2bc1a408051ba59cccc72.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/9/899210e765d5af3a1ec2bc1a408051ba59cccc72_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">904×430 31.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>It clearly has access but yet when I try installing, it doesn’t work. It gives me the same</p>
<pre><code class="lang-auto">npm ERR! remote: Repository not found.
</code></pre>
<p>error. It actually looks like the error is happening when npm is trying to run this command:</p>
<pre><code class="lang-auto">npm ERR! /usr/bin/git ls-remote -h -t https://github.com/repeaterstore/react-native-cellular-info.git
</code></pre>
<p>Apparently this command doesn’t have any output. I’ve tried it on my own machine, and I’m getting this:</p>
<pre><code class="lang-auto">$ git ls-remote -h -t https://github.com/repeaterstore/react-native-cellular-info.git
ae5f720660d1d329349ea7b4c938f3dcd22dd5fe	refs/heads/main
</code></pre>
<p>I suspected that maybe <code>npm</code> can’t find the Github registry, but I have other dependencies that ARE hosted on Github but are not private, and they install just fine.</p>
<pre><code class="lang-auto">  "dependencies": {
    "react-native-maps": "https://github.com/ArshanKhanifar/react-native-maps#29720262881b9cffd9378f92be2fbb8f28b9f09e",
  }
</code></pre>
<p>So I’m really confused why <code>npm</code> can’t find my Github project despite having access to it. Any help is greatly appreciated!</p>
<p>Thanks :heart:</p>

**Comment by lirantal** [comment]
<p>Hi <a class="mention" href="/arshankhanifar">@ArshanKhanifar</a>  :wave:</p>
<p>Hopefully this issue has been resolved, but if not then you could perhaps use the full URL to the private github repository that includes credentials, but expose them via the secrets variables, rather than hardcode:</p>
<pre><code class="lang-auto">"package-name": "git+https://&lt;github_token&gt;:x-oauth-basic@github.com/&lt;user&gt;/&lt;repo&gt;.git"
</code></pre>
<p>Another option is to also get an SSH key available on the repo/action workflow and use this syntax:</p>
<pre><code class="lang-auto">git+ssh://git@github.com:user_name/node_project.git
</code></pre>
<p>See this stackoverflow answer too: <a href="https://stackoverflow.com/questions/28728665/how-to-use-private-github-repo-as-npm-dependency" class="inline-onebox" rel="noopener nofollow ugc">node.js - How to use private Github repo as npm dependency - Stack Overflow</a></p>

**Comment by ArshanKhanifar** [comment, marked as answer]
<p>I actually tried a whole bunch of things and eventually solved this, but I think what particularly worked was making a <code>.yarnrc</code> file and adding this line to it:</p>
<pre><code class="lang-auto">"@repeaterstore:react-native-cellular-info" "https://npm.pkg.github.com"

</code></pre>

**Comment by lirantal** [comment]
<p>Gotcha. Thanks for the update!</p>

**Comment by hthetiot** [comment]
<p>Adding “persist-credentials: false” on “actions/checkout@v2” solved the issue for me.</p>
<pre><code class="lang-auto">      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v2
        with:
          persist-credentials: false
</code></pre>

**Reply by enisinanaj** [reply]
This did it for me.

**Reply by ahsansher** [reply]
Thanks man. this worked for me. 
I was stuck for 3 days. :)

**Reply by m-sirajuddin-bm** [reply]
This was perfect answer!

**Reply by ericbutz** [reply]
Yes!! this was killing me. worked great. thanks.

**Reply by Vacoris** [reply]
Was running into issue of Repository not found with a yarn install that is using the `gist+https` that included an auth and this fixed the issue. Thank you! Spent a lot of the day trying to find an explanation.

**Reply by timotismjntk** [reply]
still work even its 2025

**Reply by tonyvince** [reply]
Worked on 06.2025

**Reply by adil-darbpay** [reply]
Worked for me as well, Thanks

**Comment by jgtvares** [comment]
Can someone help me?
I'm going through this very same issue.
In my case, my private repository is in **AWS CodeArtifact** and I'm correctly authenticating to it in Github Actions.
My workflow was working 17 days ago and nothing was changed that could cause this.
I can install dependencies normally in my local setup, the problem is when running in Github Actions environment.

I have tried a thousand different things and have no clue of what could be causing this!!!
Please help!

I'm using NPM (already tried with Yarn and no success).

**UPDATE**:
Okay, solved it!
A few days ago I have deleted the upstream configuration from my **CodeArtifact** repository. I thought it was doing nothing!
So, apparently, you need an npmjs upstream for external access.
